pkg/hardwareutils/bmc/ilo4.go

```diff
@@ -63,11 +63,12 @@ func (a *iLOAccessDetails) DisableCertificateVerification() bool {
 // pre-populated with the access information, and the caller is
 // expected to add any other information that might be needed (such as
 // the kernel and ramdisk locations).
-func (a *iLOAccessDetails) DriverInfo(bmcCreds Credentials) map[string]interface{} {
+func (a *iLOAccessDetails) DriverInfo(bmcCreds Credentials, preProvExtraKernParams string) map[string]interface{} {
 	result := map[string]interface{}{
-		"ilo_username": bmcCreds.Username,
-		"ilo_password": bmcCreds.Password,
-		"ilo_address":  a.hostname,
+		"ilo_username":         bmcCreds.Username,
+		"ilo_password":         bmcCreds.Password,
+		"ilo_address":          a.hostname,
+		"kernel_append_params": preProvExtraKernParams,
```

Comment:
If this is not driver-specific (which certainly appears to be the case) then I don't feel like adding it to every driver is the right thing to do. You're already setting `kernel_append_params` later in the `Register()` call as part of `setDeployParams()`, there is no need to do it twice. So we can get rid of the change to the `DriverInfo()` API as well.